[CI] MixedClusterEsqlSpecIT test {enrich.ShadowingMultiLimit0 SYNC} failing

**Build Scans:**
- [elasticsearch-periodic #3211 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/nfawbn3ymxrus)
- [elasticsearch-periodic #3208 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/xhws4gxkbrhga)

**Reproduction Line:**
```
./gradlew ':x-pack:plugin:esql:qa:server:mixed-cluster:v8.12.2#javaRestTest' -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {enrich.ShadowingMultiLimit0 SYNC}" -Dtests.seed=B065CA6C7AD71850 -Dtests.bwc=true -Dtests.locale=mt-MT -Dtests.timezone=Canada/Yukon -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Benrich.ShadowingMultiLimit0%20SYNC%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Different column type for column [city_boundary] (GEO_SHAPE != NULL) expected:<GEO_SHAPE> but was:<NULL>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {enrich.ShadowingMultiLimit0 SYNC} (100.0% fail rate)
- [main] 2 consecutive failures in step 8.12.2_bwc (100.0% fail rate)
- [main] 2 consecutive failures in pipeline elasticsearch-periodic (100.0% fail rate)

**Note:**
This issue was created using new, in-development muting automation. Please mind any issues.

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/enrich.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/enrich.csv-spec
@@ -88,9 +88,9 @@ ROW left = "left", client_ip = "172.21.0.5", env = "env", right = "right"
 left:keyword | env:keyword | right:keyword | client_ip:keyword
 ;
 
-shadowingMulti
+shadowingMulti#[skip:-8.13.99, reason:ENRICH extended in 8.14.0]
 required_capability: enrich_load
-ROW left = "left", airport = "Zurich Airport ZRH", city = "Zürich", middle = "middle", region = "North-East Switzerland", right = "right" 
+ROW left = "left", airport = "Zurich Airport ZRH", city = "Zürich", middle = "middle", region = "North-East Switzerland", right = "right"
 | ENRICH city_names ON city WITH airport, region, city_boundary
 ;
 
@@ -98,8 +98,8 @@ left:keyword | city:keyword | middle:keyword | right:keyword | airport:text | re
 left         | Zürich       | middle         | right         | Zurich Int'l | Bezirk Zürich | "POLYGON((8.448 47.3802,8.4977 47.3452,8.5032 47.3202,8.6254 47.3547,8.5832 47.3883,8.5973 47.4063,8.5431 47.4329,8.4858 47.431,8.4691 47.4169,8.473 47.3951,8.448 47.3802))"
 ;
 
-shadowingMultiLimit0
-ROW left = "left", airport = "Zurich Airport ZRH", city = "Zurich", middle = "middle", region = "North-East Switzerland", right = "right" 
+shadowingMultiLimit0#[skip:-8.13.99, reason:ENRICH extended in 8.14.0]
+ROW left = "left", airport = "Zurich Airport ZRH", city = "Zürich", middle = "middle", region = "North-East Switzerland", right = "right"
 | ENRICH city_names ON city WITH airport, region, city_boundary
 | LIMIT 0
 ;

